python/cudf/cudf/core/column/datetime.py

```diff
@@ -229,6 +229,34 @@ def find_last_value(self, value, closest=False):
     def is_unique(self):
         return self.as_numerical.is_unique
 
+    def can_cast_safely(self, to_dtype):
+        if np.issubdtype(to_dtype, np.datetime64):
+
+            to_res, _ = np.datetime_data(to_dtype)
+            self_res, _ = np.datetime_data(self.dtype)
+
+            epoch = np.datetime64(0, to_res)
+            max_int = np.iinfo(np.dtype("int64")).max
+
+            max_dist = self.max() - epoch.astype(self.dtype)
+            min_dist = self.min() - epoch.astype(self.dtype)
```

Comment:
It feels weird that you have to run these operations to get to a timedelta type. Can you not just typecast into it zero copy?

Reply:
Apologies, I'm not sure I follow. The intent here is to find the limiting dates for the destination datatype and check the data we want to cast to make sure nothing falls outside those bounds.

Reply:
So I guess my point here is `epoch` is always guaranteed to be 0, so you end up running an unnecessary subtraction to convert `self.max()` or `self.min()` to a timedelta.

Ideally you could just do `self.max().astype(some_timedelta_type, copy=False)`

Reply:
This works and is much better, thanks!